### PR TITLE
Add support for CUDA 12 and Hopper.

### PR DESCRIPTION
**Pull request by galv**

### CODE REVIEW

**Comment by galv** [comment]
FYI. Should be a simple thing to add.

**Comment by galv** [comment]
Failing with:

```
extras/check_dependencies.sh: python2.7 is not installed
extras/check_dependencies.sh: Some prerequisites are missing; install them using the command:
  sudo apt-get install python2.7
```

I fixed this here: https://github.com/kaldi-asr/kaldi/pull/4811/commits/6d94122a6fff59ff618f75a90fb965e433b05333

But I don't want to merge that PR yet, so I will cherry-pick that commit.

**Comment by jtrmal** [comment]
LGTM, feel free to merge once you feel it's done.
The only comment I might have to adding the python2 -- I hope it won't make python2 default system "python", that would probably be very annoying.

File: .github/workflows/c-cpp.yml [hunk 1]
```diff
@@ -19,6 +19,8 @@ jobs:
     - uses: actions/checkout@v3
     - name: Install sox
       run: sudo apt-get install -y sox intel-mkl
+    - name: Install python2
+      run: sudo apt-get install -y python2
```
Comment (jtrmal) [inline .github/workflows/c-cpp.yml:23]: thanks for this

**Review by jtrmal — APPROVED** [review]

**Comment by jtrmal** [comment]
nevermind, just realized this is for the github actions VM

**Pull request merged**